# Assignment: Digital Marketing Math Part 2

### Perform step-by-step mathematical calculation to determine ROAS.

#### Dataset: Provided by Professor [Ivan Fedorenko](https://www.csueastbay.edu/directory/profiles/mktg/fedorenkoivan.html) | Digital Media Analytics - California State University, East Bay


## Objective
* You are a marketer at a financial services firm.
* You want to choose between two ad placements offered by a new publisher on their app.
* Each ad generates leads worth about **$60** each.
* Use the information provided to:
    * Compare the financial efficiency of the two placements by calculating their return on ad spend (ROAS%).
    * Decide which ad placement to buy based on this comparison.

###Given data:

* CTR A = **.003 (0.3%)**

* CTR B = **.004 (0.4%)**

* Impressions A = **2,000,000 (2M impressions)**

* Impressions B = **3,000,000 (3M impressions)**

* CPM A = **5.5 (\$5.50 Cost Per Mile)**
* CPM B = **6.0 ($6.00 Cost Per Mile)**

* Conversion rate = **0.07 (7%)**

* Revenue Per Lead (RPL) = **$60.00**

##Formula Breakdown

In [ ]:
# Step 1: Calculate Clicks
  # Clicks A = impressions A * CTR A
  # Clicks B = impressions A * CTR B

# Step 2: Calculate Leads
  # Leads A = Clicks A * Conversion rate
  # Leads B = Clicks B * Conversion rate

# Step 3: Calculate Revenue
  # Revenue A = Leads A * RPL
  # Revenue B = Leads B * RPL

# Step 4: Calculate Ad Cost
  # Cost A = (Impressions A / 1000) * CPM A
  # Cost B = (Impressions B / 1000) * CPM B

# # Step 5: Calculate ROAS (%)
  # ROAS A = (Revenue A / Cost A) * 100
  # ROAS B = (Revenue B / Cost B) * 100

## Mathematical Calculations:

In [ ]:
# Step 1: Calculate Clicks
clicks_a = 2000000 * 0.003
clicks_b = 3000000 * 0.004

clicks_a, clicks_b

(6000.0, 12000.0)

In [ ]:
# Step 2: Calculate Leads
leads_a = clicks_a * 0.07
leads_b = clicks_b * 0.07

leads_a, leads_b

(420.00000000000006, 840.0000000000001)

In [ ]:
# Step 3: Calculate Revenue
revenue_a = leads_a * 60
revenue_b = leads_b * 60

revenue_a, revenue_b

(25200.000000000004, 50400.00000000001)

In [ ]:
# Step 4: Calculate Ad Cost
cost_a = (2000000 / 1000) * 5.5
cost_b = (3000000 / 1000) * 6.0

cost_a, cost_b

(11000.0, 18000.0)

In [ ]:
#Step 5: Calculate ROAS (Return on Ads Spend)
roas_a = (revenue_a / cost_a) * 100
roas_b = (revenue_b / cost_b) * 100

roas_a, roas_b

(229.09090909090915, 280.0)

## Recommendation:
Since **Placement B** has a higher ROAS (**280%**) compared to Placement A (229.1%), **Placement B** is the **better investment** for ad placement.